### ISSUE TITLE
Dark icon

### ISSUE THREAD

**Issue by happysurf** [jingyu9575/multithreaded-download-manager#95]
In the Firefox addons dark page, the MDM icon is hard to see, will be great make a medium gray or something like that. 

![Dark Icon](https://user-images.githubusercontent.com/13637587/66698297-fe01c580-ecdc-11e9-87be-b7a12933d4a7.png)

**Comment by jingyu9575** [comment]
There is no way to specify two icons for light/dark themes, and the black icon is choosen to follow the style of built-in extensions. 

How did you apply the dark theme to this page? The built-in dark theme does not affect about:addons.

**Comment by happysurf** [comment]
For apply the dark theme set **browser.in-content.dark-mode** to **true** (default in Firefox 70 beta).
In any case the built-in icon color is not only black, but there are many option to select in the options.
For that the icon color in the add-ons page could be different to toolbar customized color.

![Icon](https://user-images.githubusercontent.com/13637587/66731782-52ca4b00-ee59-11e9-95c2-2592a2aa0e4a.png)

**Comment by jingyu9575** [comment]
With dark theme and this pref set to true, the about:addon page is still light on my Firefox Nightly latest. Are you using userContent.css? How does the built-in icons such as Flash Player and Default Theme show?

The color option only applies to the toolbar icon. The addon icon in about:addons is a static property and cannot be changed in the program.

**Comment by happysurf** [comment]
> 
>With dark theme and this pref set to true, the about:addon page is still light on my Firefox Nightly latest.

I don't know why but is the same for other users.

> 
>Are you using userContent.css?

No.

> 
>How does the built-in icons such as Flash Player and Default Theme show?

![Icons](https://user-images.githubusercontent.com/13637587/66893109-96ee5480-efed-11e9-820b-5094ce4177d6.png)

> 
>The color option only applies to the toolbar icon. The addon icon in about:addons is a static property and cannot be changed in the program.

I know, in fact you must decide a dark gray color IMHO.

**Comment by jingyu9575** [comment]
Please set `svg.context-properties.content.enabled` to `true`. This is required to apply context colors to third-party icons.

**Comment by happysurf** [comment]
In the last Firefox 70 build2, the toolbar icon no more change the color in light or white thru the interface options but remain black.
http://ftp.mozilla.org/pub/firefox/candidates/70.0-candidates/

**Comment by jingyu9575** [comment]
> the toolbar icon no more change the color

The code is accidentally broken in v2.3. Fixed in v2.3.1.

> the add-on page icon is black

This one? Have you set the pref `svg.context-properties.content.enabled`?

![图片](https://user-images.githubusercontent.com/8923139/67089432-1e002000-f1da-11e9-8179-31c3af1853a9.png)

**Comment by happysurf** [comment]
With 2.3.1 the toolbar icon problem is fixed, but unfortunately the add-on page icon is still black with **svg.context-properties.content.enabled** to true.

**Comment by jingyu9575** [comment]
Please right-click the icon and select "inspect element". What's the element code (`<img ...>`)?

![图片](https://user-images.githubusercontent.com/8923139/67112536-08a2ea00-f20a-11e9-873f-f99de92481ed.png)

**Comment by happysurf** [comment]
![Code](https://user-images.githubusercontent.com/13637587/67112997-ca410700-f1d8-11e9-8543-c374faf59427.png)

**Comment by jingyu9575** [comment]
Your about:addons is using the icon from [addons.mozilla.org](https://addons.mozilla.org/zh-CN/firefox/addon/multithreaded-download-manager/), which is only black. The new icon packed in the extension is not being used. 
I'm not sure why Firefox is picking this online icon. I'll look into it.

**Comment by happysurf** [comment]
This morning in my office PC, with same Firefox configuration, the MDM add-ons page icon is white.
This afternoon I will try to reinstall the extension.

![Icons4](https://user-images.githubusercontent.com/13637587/67181496-1361a880-f3dd-11e9-9de0-88ba764c2166.png)

**Comment by jingyu9575** [comment]
Yes this `src="jar:file:///..."` is the correct local icon packed in the extension. The `src="https://addons.cdn.mozilla.net/..."` is the black one on the website.

I'm still not sure what causes Firefox to use the local or web icons. Right now on my system, Firefox prefers the local icon if it exists, and only use the web icon if the extension does not have icon packed.

**Comment by happysurf** [comment]
Unfortunately on my netbook also after remove and reinstall MDM, the icon is still taken from Mozilla web site.
